src/main/java/frc/robot/subsystems/vision/VisionSubsystem.java

```diff
@@ -126,7 +145,54 @@ public PoseEstimate getMegaTag1PoseEstimate(int limelightNumber) {
    *     return 0 for x, y, and theta
    */
   public PoseEstimate getMegaTag2PoseEstimate(int limelightNumber) {
-    return LimelightHelpers.getBotPoseEstimate_wpiBlue_MegaTag2(getLimelightName(limelightNumber));
+    try {
+      return LimelightHelpers.getBotPoseEstimate_wpiBlue_MegaTag2(getLimelightName(limelightNumber));
+    } catch (Exception e) {
+        System.err.println("Error estimating MegaTag2 pose for the: " + getLimelightName(limelightNumber) + e.getMessage());
+        return new PoseEstimate(); // Return an empty PoseEstimate
+    }
+  }
+
+  /**
+   * Checks is megatag 1 and megatag 2 estimates are good
+   * @param limelightNumber
+   * @return
+   */
+  public boolean isValidPoseEstimate(int limelightNumber) {
```

Comment:
What are your thoughts on making it so if there is a big discrepancy between mt1 and mt2, we just go with mt1 rather than not doing any pose estimation?

Reply:
This one just checks if both of them are within the field

Reply:
I think the current plan is if there is a large discrepancy we will use mt1